### ISSUE TITLE
Any OPTIONS request causes "MethodError: merge" (needed for CORS)

### ISSUE THREAD

**Issue by Helveg** [essenciary/Genie.jl#119]
Whenever I make an OPTIONS request I get a 500 Internal Server Error:

```
Distributed.RemoteException(1, CapturedException(MethodError(merge, (Pair{SubString{String},SubString{String}}["Server"=>"Genie/0.7.0/Julia/1.0.0", "Content-Type"=>"text/html"], Dict("Access-Control-Allow-Headers"=>"","Access-Control-Allow-Methods"=>"","Access-Control-Allow-Credentials"=>"","Access-Control-Allow-Origin"=>"","Access-Control-Expose-Headers"=>"","Access-Control-Max-Age"=>"86400")), 0x0000000000006225), Any[(set_headers!(::Request, ::Response, ::Response) at AppServer.jl:110, 1), (handle_request(::Request, ::Response, ::IPv4) at AppServer.jl:96, 1), (handle_request(::Request, ::Response) at AppServer.jl:95, 1), (#14 at macros.jl:108 [inlined], 1), (run_work_thunk(::##14#17{Request,Response}, ::Bool) at process_messages.jl:56, 1), (#remotecall_fetch#148(::Pairs{Union{},Union{},Tuple{},NamedTuple{(),Tuple{}}}, ::Function, ::Function, ::LocalProcess) at remotecall.jl:364, 1), (remotecall_fetch(::Function, ::LocalProcess) at remotecall.jl:364, 1), (#remotecall_fetch#152(::Pairs{Union{},Union{},Tuple{},NamedTuple{(),Tuple{}}}, ::Function, ::Function, ::Int64) at remotecall.jl:392, 1), (remotecall_fetch(::Function, ::Int64) at remotecall.jl:392, 1), (setup_http_handler(::Request, ::Response) at AppServer.jl:60, 1), (setup_http_handler at AppServer.jl:59 [inlined], 1), (#6 at AppServer.jl:40 [inlined], 1), (handle at Handlers.jl:248 [inlined], 1), (handle(::RequestHandlerFunction{##6#12}, ::Stream{Request,Transaction{TCPSocket}}) at Handlers.jl:271, 1), (#4 at Handlers.jl:339 [inlined], 1), (macro expansion at Servers.jl:352 [inlined], 1), ((::##13#14{##4#5{RequestHandlerFunction{##6#12}},Transaction{TCPSocket},Stream{Request,Transaction{TCPSocket}}})() at task.jl:259, 1)])) in C:\Users\Robin\.julia\packages\Genie\AxLFn\src\AppServer.jl:71




######################### ERROR STACKTRACE #########################
Distributed.RemoteException(1, CapturedException(MethodError(merge, (Pair{SubString{String},SubString{String}}["Server"=>"Genie/0.7.0/Julia/1.0.0", "Content-Type"=>"text/html"], Dict("Access-Control-Allow-Headers"=>"","Access-Control-Allow-Methods"=>"","Access-Control-Allow-Credentials"=>"","Access-Control-Allow-Origin"=>"","Access-Control-Expose-Headers"=>"","Access-Control-Max-Age"=>"86400")), 0x0000000000006225), Any[(set_headers!(::Request, ::Response, ::Response) at AppServer.jl:110, 1), (handle_request(::Request, ::Response, ::IPv4) at AppServer.jl:96, 1), (handle_request(::Request, ::Response) at AppServer.jl:95, 1), (#14 at macros.jl:108 [inlined], 1), (run_work_thunk(::##14#17{Request,Response}, ::Bool) at process_messages.jl:56, 1), (#remotecall_fetch#148(::Pairs{Union{},Union{},Tuple{},NamedTuple{(),Tuple{}}}, ::Function, ::Function, ::LocalProcess) at remotecall.jl:364, 1), (remotecall_fetch(::Function, ::LocalProcess) at remotecall.jl:364, 1), (#remotecall_fetch#152(::Pairs{Union{},Union{},Tuple{},NamedTuple{(),Tuple{}}}, ::Function, ::Function, ::Int64) at remotecall.jl:392, 1), (remotecall_fetch(::Function, ::Int64) at remotecall.jl:392, 1), (setup_http_handler(::Request, ::Response) at AppServer.jl:60, 1), (setup_http_handler at AppServer.jl:59 [inlined], 1), (#6 at AppServer.jl:40 [inlined], 1), (handle at Handlers.jl:248 [inlined], 1), (handle(::RequestHandlerFunction{##6#12}, ::Stream{Request,Transaction{TCPSocket}}) at Handlers.jl:271, 1), (#4 at Handlers.jl:339 [inlined], 1), (macro expansion at Servers.jl:352 [inlined], 1), ((::##13#14{##4#5{RequestHandlerFunction{##6#12}},Transaction{TCPSocket},Stream{Request,Transaction{TCPSocket}}})() at task.jl:259, 1)])) in C:\Users\Robin\.julia\packages\Genie\AxLFn\src\AppServer.jl:71


```

**Comment by essenciary** [comment]
@Helveg Thanks for letting me know - I'll check it out. 
Also, probably won't fix, but it looks like you're not on the latest Genie (nor the latest Julia). It should be in the line of "Server"=>"Genie/0.8.2/Julia/1.1.0"

**Comment by essenciary** [comment]
@Helveg Fixed - available in the newly pushed Genie v0.8.3. 

You can either set up an OPTIONS route and handle the request yourself (maybe using the "*" route to catch all OPTION requests) — or do nothing and Genie will respond automatically with the `Configuration.Settings.cors_headers` Dict (which you can overwrite)

**Comment by Helveg** [comment]
Thanks!
